### ISSUE TITLE
On macOS the frame size could change without window resize

### ISSUE THREAD

**Issue by kchibisov** [rust-windowing/winit#2191]
If you're using multiple windows and do tabbing for them the the of the frame could change without size of the window being changed, leading to winit not issuing resize events for the windows in tab group. So winit should send resize event when the frame changes as well, and not only when the window gets resized.

I'm not familiar with macOS API, however I could say that the frame rect we get in https://github.com/rust-windowing/winit/blob/0e52672f4aea376c3ef8a06b64aedca46a40e9b8/src/platform_impl/macos/view.rs#L324 is absolutely correct. And `window_did_resize` method on window delegate don't get called at all.

@madsmtm Do you known something about it? Seems like something that could be easily fixed?

**Comment by sshock** [comment]
(Note: to test tabbed windows in macOS, make sure to set "Prefer tabs" to `always` in system preferences.)

I was able to confirm using the `multiwindow` example that hitting a key to create a new (tabbed) window triggers two `frameDidChange` and does not trigger `windowDidResize` (and therefore does not emit a `WindowEvent::Resized`).

However, here is the interesting part: if after creating one or more (tabbed) windows I use the mouse to resize the window, then after that point every new (tabbed) window created will trigger all four of these:
* frameDidChange
* frameDidChange
* windowDidResize
* frameDidChange

As for the rect size, I have noticed while testing with `multiwindow` in glutin that when resizing explicitly with the mouse the size is correct, but the one emitted when creating a new (tabbed) window has a height that is a little too big (like it's including the height of the tab).

The good news is it sounds like if we emit resize event from frame_did_change then not only will it fix the problem of it not getting emitted all the time, but it should have the correct rect.

It does seem though that even after this is fixed in some scenarios we will get multiple resize events, some with the wrong rect and some with the correct.  But I guess that is ok as long as the one with the correct one comes last?  (🤞)
